src/core/reference/include/ngraph/runtime/reference/grid_sample.hpp

```diff
@@ -0,0 +1,300 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <array>
+#include <cassert>
+#include <cfenv>
+#include <cmath>
+
+#include "ngraph/shape.hpp"
+#include "openvino/op/grid_sample.hpp"
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+namespace {
+
+using index_4D_t = typename std::array<size_t, 4>;
+
+template <typename GRID_ET>
+using denormalize_fn_t = typename std::function<GRID_ET(GRID_ET, size_t)>;
+
+template <typename DATA_ET>
+using get_padded_fn_t = typename std::function<DATA_ET(const DATA_ET*, const Shape&, size_t, size_t, long, long)>;
+
+template <typename T>
+T& get_v(T* buffer, const Shape& shape, const index_4D_t& index) {
```

Comment:
`get_v` doesn't say much about what this function does. Can you rename it?

Reply:
Done.